File: Assets/Scripts/ScrollUVFar.cs
using UnityEngine;
using System.Collections;

public class ScrollUVFar : MonoBehaviour {

	// Update is called once per frame
	void Update () {
		Material mat = this.renderer.material;
		Vector2 offset = mat.mainTextureOffset;

		offset.x = transform.position.x / 300;
		offset.y = transform.position.y / 300;
		
		mat.mainTextureOffset = offset;
	}
}

File: Assets/Scripts/ScrollUVFar.js
#pragma strict

function Update () {
	var mat = this.renderer.material;
	var offset : Vector2 = mat.mainTextureOffset;
	
	
	offset.x = transform.position.x / 300;
	offset.y = transform.position.y / 300;
	
	mat.mainTextureOffset = offset;
}